.github/workflows/app-publish.yaml

```diff
@@ -0,0 +1,121 @@
+name: App Publish
+
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - release*
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    if: ${{ startsWith(github.ref, 'refs/tags/v') || github.repository == 'jellyfin/jellyfin-android' }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Setup Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
```

Comment:
Don't use caching for the publish task, always build clean so the cache can't cause or suppress any issues.

Reply:
Sounds reasonable, I have not thought about it this way NGL